File: src/main/resources/static/code/counter/Application.java
@AppTitle("Simple Counter")
public final class Application extends App {

  private int count = 0;
  Paragraph text = new Paragraph("Count: 0");
  Button button = new Button("Counter");

  @Override
  public void run() throws WebforjException {

    Frame mainFrame = new Frame();
    mainFrame.add(text, button);

    button.onClick(e -> {
      count++;
      text.setText("Count: " + count);
    });
  }
}

File: src/main/resources/static/code/counter/Application.kt
@AppTitle("Simple Counter")
class Application : App() {
  private var count = 0
  private val text = Paragraph("Count: 0")
  private val button = Button("Counter")

  override fun run() {
    val mainFrame = Frame()
    mainFrame.add(text, button)

    button.onClick {
      count++
      text.text = "Count: $count"
    }
  }
}